common_design.theme

```diff
@@ -238,3 +285,43 @@ function common_design_preprocess_menu(&$variables, $hook) {
     $variables['#cache']['contexts'][] = 'user';
   }
 }
+
+/**
+ * Render a block.
+ *
+ * We cache a flag indicating the block has already been redendered and return
+ * an empty array in that case to ensure that the block is displayed only one
+ * on the page.
+ *
+ * @param string $id
+ *   Block ID.
+ * @param bool $translate
+ *   Translate the block or not. This allows to keep some elements of the page
+ *   in the site's default language for example like the local tasks (view, edit
+ *   etc.).
+ *
+ * @return array
+ *   Renderable array of the block. Empty array if this was already called
+ *   before to avoid displaying the block several times.
+ *
+ * @see https://drupal.stackexchange.com/a/171733
+ */
+function common_design_get_block_render_array($id, $translate = TRUE) {
+  $rendered = &drupal_static('common_design_rendered_blocks');
+
+  if (!isset($rendered[$id])) {
+    // Prevent rendering the block several times.
+    $rendered[$id] = TRUE;
+
+    // Generate an instance of the plugin block.
+    $block_manager = \Drupal::service('plugin.manager.block');
+    $plugin_block = $block_manager->createInstance($id, []);
+
+    // Skip if there is no block matching the id.
+    if (empty($plugin_block)) {
+      return [];
+    }
+  }
+
+  return [];
+}
```

Comment:
@orakili Can you inspect this to double check only relevant code is included? I copied from SLT and GHO and removed what I thought wasn't needed.

Reply:
I'll also add the access check from https://github.com/UN-OCHA/slt8-site/blob/develop/html/themes/custom/common_design_subtheme/common_design_subtheme.theme#L170